solana/programs/example-native-token-transfers/src/instructions/admin.rs

```diff
@@ -15,13 +15,19 @@ use crate::{
 
 // * Transfer ownership
 
-/// Transferring the ownership is a 2-step process. The first step is to set the
+/// For safety reasons, transferring ownership is a 2-step process. The first step is to set the
 /// new owner, and the second step is for the new owner to claim the ownership.
 /// This is to prevent a situation where the ownership is transferred to an
 /// address that is not able to claim the ownership (by mistake).
 ///
 /// The transfer can be cancelled by the existing owner invoking the [`claim_ownership`]
 /// instruction.
+///
+/// Alternatively, the ownership can be transferred in a single step by calling the
+/// [`transfer_ownership_one_step`] instruction. This can be dangerous because if the new owner
+/// cannot actually sign transactions (due to setting the wrong address), the program will be
+/// permanently locked. If the intention is to transfer ownership to a program using this instruction,
+/// take extra care to ensure that the owner is a PDA, not the program address itself.
```

Comment:
Is there anything we can do to help the user here? Could we check that the account has data for example and use that as a heuristic about whether that series of bytes should _really_ be the permanent owner?

Perhaps one approach would be adding extra endpoints here, like `transfer_owner_to_account` vs. `transfer_owner_to_program_derived_account`? I'm asking in case there are any checks we can do to help the user not brick the owner here. If they use one route we could help them determine whether the address is really a PDA or else a regular user account and so on.

Arguably any of these would be a restriction especially if the point of this entrypoint is to do this cross-chain basically not worry about errors.

Reply:
yeah, unfortunately we can't really tell if it's a sensible account or not. The only way to tell is if the new account demonstrates that it can sign a transaction, which is why the existing flow is 2-step. This is the "I know what I'm doing" use case, so everyone should use it at their own peril

Reply:
Fair point, I suppose it does sort of put pressure on someone to take the normal, safe path too if we provide absolutely no help on the dangerous route